Files/warranty_dell.py

```diff
@@ -77,114 +137,101 @@ def process_result(self, result, purchases):
         global full_serials
         data = {}
 
-        if 'AssetWarrantyResponse' in result:
-            for item in result['AssetWarrantyResponse']:
-                try:
-                    warranties = item['AssetEntitlementData']
-                    asset = item['AssetHeaderData']
-                    product = item['ProductHeaderData']
-                except IndexError:
-                    if self.debug:
-                        try:
-                            msg = str(result['InvalidFormatAssets']['BadAssets'])
-                            if msg:
-                                print '\t\t[-] Error: Bad asset: %s' % msg
-                        except Exception as e:
-                            print e
+        for item in result:
+            try:
+                warranties = item['entitlements']
+            except IndexError:
+                if self.debug:
+                    try:
+                        msg = str(result['InvalidFormatAssets']['BadAssets'])
+                        if msg:
+                            print '\t\t[-] Error: Bad asset: %s' % msg
+                    except Exception as e:
+                        print e
 
+            else:
```

Comment:
This "else" does not seem to be properly lined up with an "if".

Reply:
this is a try-except-else clause. was confused about this too but it seems it is an alternative to finally that is run before all finally statements. Previous version was written like this